go/libraries/doltcore/diff/diff_summary.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package diff
+
+import (
+	"context"
+	"errors"
+
+	"github.com/liquidata-inc/dolt/go/store/atomicerr"
+	"github.com/liquidata-inc/dolt/go/store/diff"
+	"github.com/liquidata-inc/dolt/go/store/types"
+)
+
+type diffFunc func(changeChan chan<- types.ValueChanged, stopChan <-chan struct{})
+
+type DiffSummaryProgress struct {
+	Adds, Removes, Changes, CellChanges, NewSize, OldSize uint64
+}
+
+// Summary reports a summary of diff changes between two values
+func Summary(ctx context.Context, ae *atomicerr.AtomicError, ch chan DiffSummaryProgress, v1, v2 types.Map) {
+	if !v1.Equals(v2) {
+		if diff.ShouldDescend(v1, v2) {
```

Comment:
Can ShouldDescend ever be false for 2 maps?  If not I'd probably just remove the if statement, and put the logic of diffSummaryValueChanged in here directly.

Reply:
No it shouldn't - took out the if statement and combined those functions [4da04bc](https://github.com/liquidata-inc/dolt/pull/105/files#diff-f7ee563c78a9fb4951bf98eae945d4bcR32)